end-to-end-untargeted-metabolomics.Rmd

```diff
@@ -383,70 +398,94 @@ eic_Met <- chromatogram(data, rt = c(170, 200),
                            mz = Met_mz + c(-0.05, 0.05), aggregationFun = "max")
 ```
 
-Now we can `plot()` these EICs: 
+Now we can `plot()` these EICs:
 
 ```{r plot EIC, echo=FALSE}
 
 #' plot both EIC
 par(mfrow = c(1, 2))
-plot(eic_cystine, main = "Cystine", col = paste0(col_phenotype[bpc$phenotype], 80))
+plot(eic_cystine, main = "Cystine", col = paste0(col_sample, 80))
 grid()
 
-plot(eic_Met, main = "1- Methylhistidine", col = paste0(col_phenotype[bpc$phenotype], 80))
+plot(eic_Met, main = "1- Methylhistidine", col = paste0(col_sample, 80))
 grid()
 legend("topright", col = col_phenotype,
        legend = names(col_phenotype), lty = 1)
 ```
 
-... describe EICs
+... describe EICs - for cytine there is a clear concentration difference between
+QCs and study samples.
 
 ## Internal standard chromatographich data (don't keep that for final version)
 
 ```{r EIC extract for internal standard, eval=FALSE, include=FALSE}
-#' get the list 
-intern_standard <- read.delim("C:/Users/plouail/Documents/GitHub/lcms-standards/data/internal_standards.txt") 
-intern_standard <- intern_standard[-4, ] ## remove succinic acid because no POS
+#' get the list
+intern_standard <- read.delim("internal_standards.txt")
+intern_standard <- intern_standard[!is.na(intern_standard$POS), ]
 
 #'generate calcualte formula
-intern_standard[, "mz"] <- mass2mz(intern_standard$formula, adduct = "[M+H]+")[, 1]
-intern_standard[3, "mz"] <- mass2mz(intern_standard[3,"formula"], adduct = "[M+Na]+")[, 1]
+intern_standard$mz <- mapply(intern_standard$formula,
```

Comment:
just a suggestion - to support also more complicated configurations (e.g. different adducts for different compounds). look over both formula and adduct definition and calculate just that adduct value. Original code might be computationally more efficient though.